Added mobilefacedet-v1-mxnet model (#1010)
opencv/open_model_zoo: models/public/mobilefacedet-v1-mxnet/mobilefacedet-v1-mxnet.md

# mobilefacedet-v1-mxnet

## Use Case and High-Level Description

>   MobileFace Detection V1 is a Light and Fast Face Detector for Edge Devices (LFFD) model based on Yolo V3 architecture and trained with MXNet. For details see the [repository](https://github.com/becauseofAI/MobileFace) and [paper](https://arxiv.org/pdf/1904.10633.pdf).

Reviewer: ```suggestion
  MobileFace Detection V1 is a Light and Fast Face Detector for Edge Devices (LFFD) model based on Yolo V3 architecture and trained with MXNet\*. For details see the [repository](https://github.com/becauseofAI/MobileFace) and [paper](https://arxiv.org/pdf/1904.10633.pdf).
```

Author: Done

## Specification

| Metric            | Value         |
|-------------------|---------------|
| Type              | Detection     |
| GFLOPs            | 3.5456       |
| MParams           | 7.6828        |
| Source framework  | MXNet\*       |

## Input

### Original model

> Image, name - `data`, shape - `[1x256x256x3]`, format -`[BxHxWxC]` where:

Reviewer: I see, the BHWC->BCHW conversion only applies to TF model, am i right?

Reviewer 2: Model contains preprocessing node inside which transpose NHWC->NCHW  and it is not possible to remove (I asked MO team about it, they said no way)

Author: The original model is not converted to BCHW format. This model takes input in a  BHWC format and converts the input data within itself.

Reviewer: Ok, fine. But I strongly recommend you to add description of converted model input with some 
>**WARNING**

sign, because most of models n OMZ has NCHW layout

Author: please review again

- `B` - batch size
- `H` - height
- `W` - width
- `C` - channel

Expected color order -  `BGR`.

### Converted model

The converted model has the same parameters as the original model.

## Output

### Original model

1. The array of detection summary info, name - `yolov30_slice_axis1`,  shape - `1,18,8,8`. The anchor values are `118,157,  186,248,  285,379`.

2. The array of detection summary info, name - `yolov30_slice_axis2`,  shape - `1,18,16,16`. The anchor values are `43,54,  60,75,  80,106`.

3. The array of detection summary info, name - `yolov30_slice_axis3`,  shape - `1,18,32,32`. The anchor values are `10,12,  16,20,  23,29`.

For each case format is `B,N,Cx,Cy`, where
    - `B` - batch size
    - `N` - number of detection boxes for cell

>     - `Cx`, `Cy` - cell index

Reviewer: More correct format is `B,3*BB, Cx,Cy`, where `BB` is a bounding box for face.
You can use any other name instead of `BB`.

Author: what is 3 means?

Reviewer: Model produces three bounding boxes with according anchors for each cell. 
(so, true format is `B,N,BB,CX,CY`, where N=3 (bbox count), BB=6 (x,y,h,w,conf,class) etc.)

Reviewer: See as [example](https://github.com/opencv/open_model_zoo/blob/develop/models/public/yolo-v3-tf/yolo-v3-tf.md)

Author: please review again

Detection box has format [`x`,`y`,`h`,`w`,`conf`,`class`], where:
- (`x`,`y`) - coordinates of box center, relative to cell
- `h`,`w` - normalized height and width of box
- `conf` - confidence of detection box
- `class` - score for class

### Converted model

1. The array of detection summary info, name - `yolov30_yolooutputv30_conv0_fwd/YoloRegion`,  shape - `1,18,8,8`. The anchor values are `118,157,  186,248,  285,379`.

2. The array of detection summary info, name - `yolov30_yolooutputv31_conv0_fwd/YoloRegion`,  shape - `1,18,16,16`. The anchor values are `43,54,  60,75,  80,106`.

3. The array of detection summary info, name - `yolov30_yolooutputv32_conv0_fwd/YoloRegion`,  shape - `1,18,32,32`. The anchor values are `10,12,  16,20,  23,29`.

For each case format is `B,N,Cx,Cy`, where
    - `B` - batch size
    - `N` - number of detection boxes for cell
    - `Cx`, `Cy` - cell index

Detection box has format [`x`,`y`,`h`,`w`,`conf`,`class`], where:
- (`x`,`y`) - coordinates of box center, relative to cell
- `h`,`w` - normalized height and width of box
- `conf` - confidence of detection box

> - `class` - score for class in the [0,1] range

Reviewer: What the difference between `score` and `confidence` ?

Reviewer 2: It is not clear for me, the reason of this question... Could you please give more context?

Reviewer 2: generally, `conf` is objectness score (there is object in this cell or not), `class` is confidence that detected object belongs to specific class.
The output format is the same like in Yolo V3, but model trained to detect objects for only 1 class - face

Reviewer: Yes, I understand the difference, and even why here present both of them. But for the outside view it is not so obvious and may be confusing.

Also, do we need some postprocessing, like multiplying them to get correct score (like YOLO do)?

Reviewer 2: @eizamaliev if I right understand, @Anna-Mironova just  tried to follow guidelines given in yolo-v3 model description, preserving the same info level and terminology (BTW, yolo-v3 description does not contains info about postprocessing). If you have suggestion how to improve it, please give. We will be appreciate to you help.

Reviewer: @eaidova thanks for remark. I will expand descriptions of YOLOs and then help you.

Reviewer: Please refer to [PR 1029](https://github.com/opencv/open_model_zoo/pull/1029) (also you are welcomed for review)

Reviewer: My main point was what `conf`  is a probability of object (any type) and `classes` is a probability distribution over classes. And the real class confidence is multiplication of both. But for only one class it is pointless.

Author: @eizamaliev , @eaidova  thanks. please review again

# mobilefacedet-v1-mxnet

## Use Case and High-Level Description

  MobileFace Detection V1 is a Light and Fast Face Detector for Edge Devices (LFFD) model
  based on Yolo V3 architecture and trained with MXNet.
  For details see the [repository](https://github.com/becauseofAI/MobileFace)

>   and [paper](https://arxiv.org/pdf/1904.10633.pdf)

Author: Done

>   and [paper](https://arxiv.org/pdf/1904.10633.pdf)

Author: Done

>   and [paper](https://arxiv.org/pdf/1904.10633.pdf)

Reviewer 3: It still is, though?

>   and [paper](https://arxiv.org/pdf/1904.10633.pdf)

Author: Done

# mobilefacedet-v1-mxnet

## Use Case and High-Level Description

  MobileFace Detection V1 is a Light and Fast Face Detector for Edge Devices (LFFD) model based on Yolo V3 architecture and trained with MXNet\*. For details see the [repository](https://github.com/becauseofAI/MobileFace) and [paper](https://arxiv.org/pdf/1904.10633.pdf).

## Specification

| Metric            | Value         |
|-------------------|---------------|
| Type              | Detection     |
| GFLOPs            | 3.5456       |
| MParams           | 7.6828        |
| Source framework  | MXNet\*       |

## Input

### Original model

Image, name - `data`, shape - `[1x256x256x3]`, format -`[BxHxWxC]` where:

- `B` - batch size
- `H` - height
- `W` - width
- `C` - channel

Expected color order -  `BGR`.

### Converted model

The converted model has the same parameters as the original model.

>     WARNING: Note that input layout of converted model is `[BxHxWxC]`.

Reviewer 2: I think you need to highlight it using markdown features. This  appeal sound rude. Possibly we can make it more soft if it will start with `Please note` or `Please pay attention` instead of  just Note

Author: please review again

> > **WARNING:** Please pay attention that input layout of converted model is `[BxHxWxC]`.

Reviewer 3: ```suggestion
> **WARNING:** Please note that the input layout of the converted model is `[BxHxWxC]`.
```

You can only pay attention _to_ something, not _that_ something happens. It's possible to contort the sentence to keep "pay attention" in it (e.g. "pay attention to the fact that the input layout is [...]"), but it would be simpler to just use "note".

Author: Done

2. The array of detection summary info, name - `yolov30_slice_axis2`,  shape - `1,18,16,16`. The anchor values are `43,54,  60,75,  80,106`.

3. The array of detection summary info, name - `yolov30_slice_axis3`,  shape - `1,18,32,32`. The anchor values are `10,12,  16,20,  23,29`.

For each case format is `B,N*DB,Cx,Cy`, where
    - `B` - batch size
    - `N` - number of detection boxes for cell
    - `DB` - size of each detection box
    - `Cx`, `Cy` - cell index

Detection box has format [`x`,`y`,`h`,`w`,`box_score`,`face_score`], where:
- (`x`,`y`) - raw coordinates of box center, apply [sigmoid function](https://en.wikipedia.org/wiki/Sigmoid_function) to get coordinates relative to the cell
- `h`,`w` - raw height and width of box, apply [exponential function](https://en.wikipedia.org/wiki/Exponential_function) and multiply by corresponding anchors to get height and width values relative to cell
- `box_score` - confidence of detection box, apply [sigmoid function](https://en.wikipedia.org/wiki/Sigmoid_function) to get confidence in [0,1] range

> - `face_score` - probability that detected object belongs to specific class, apply [sigmoid function](https://en.wikipedia.org/wiki/Sigmoid_function) to get confidence in [0,1] range

Reviewer: ```suggestion
- `face_score` - probability that detected object belongs to face class, apply [sigmoid function](https://en.wikipedia.org/wiki/Sigmoid_function) to get confidence in [0,1] range
```

Author: Done

2. The array of detection summary info, name - `yolov30_yolooutputv31_conv0_fwd/YoloRegion`,  shape - `1,18,16,16`. The anchor values are `43,54,  60,75,  80,106`.

3. The array of detection summary info, name - `yolov30_yolooutputv32_conv0_fwd/YoloRegion`,  shape - `1,18,32,32`. The anchor values are `10,12,  16,20,  23,29`.

For each case format is `B,N*DB,Cx,Cy`, where
    - `B` - batch size
    - `N` - number of detection boxes for cell
    - `DB` - size of each detection box
    - `Cx`, `Cy` - cell index

Detection box has format [`x`,`y`,`h`,`w`,`conf`,`class`], where:
- (`x`,`y`) - raw coordinates of box center, apply [sigmoid function](https://en.wikipedia.org/wiki/Sigmoid_function) to get coordinates relative to the cell
- `h`,`w` - raw height and width of box, apply [exponential function](https://en.wikipedia.org/wiki/Exponential_function) and multiply by corresponding anchors to get height and width values relative to cell
- `conf` - confidence of detection box, use [sigmoid function](https://en.wikipedia.org/wiki/Sigmoid_function) to get confidence in [0,1] range

> - `class` - probability that detected object belongs to specific class, use [sigmoid function](https://en.wikipedia.org/wiki/Sigmoid_function) to get confidence in [0,1] range

Reviewer: Are you sure, that sigmoid is necessary for **converted** model? AFAIK, `TFYOLOV3` transformations already apply some postprocessing.
